Fill.py

```diff
@@ -929,133 +906,153 @@ def failed(warning: str, force: typing.Union[bool, str]) -> None:
-            elif 'locations' in block:
-                locations = block['locations']
+            locations: typing.List[str] = block.locations
             if isinstance(locations, str):
                 locations = [locations]
+            elif not isinstance(locations, list):
+                raise Exception(f"Plando 'locations' has to be a list, not {type(locations)} for player {player}.")
+
+            locations_from_groups: typing.List[str] = []
+            resolved_locations: typing.List[Location] = []
+            for target_player in worlds:
+                world_locations = multiworld.get_locations(target_player)
+                for group in multiworld.worlds[target_player].location_name_groups:
+                    if group in locations:
+                        locations_from_groups.extend(multiworld.worlds[target_player].location_name_groups[group])
+                resolved_locations.extend(location for location in world_locations
+                                          if location.name in [*locations, *locations_from_groups])
+            new_block["locations"] = sorted(dict.fromkeys(locations))
+            new_block["resolved_locations"] = sorted(set(resolved_locations))
+
+            new_block["count"] = block.count
+            if not new_block["count"]:
+                new_block["count"] = (min(len(new_block["items"]), len(new_block["locations"])) if
+                                      len(new_block["locations"]) > 0 else len(new_block["items"]))
+            if isinstance(new_block["count"], int):
+                new_block["count"] = {"min": new_block["count"], "max": new_block["count"]}
+            if "min" not in new_block["count"]:
+                new_block["count"]["min"] = 0
+            if "max" not in new_block["count"]:
+                new_block["count"]["max"] = (min(len(new_block["items"]), len(new_block["locations"])) if
+                                             len(new_block["locations"]) > 0 else len(new_block["items"]))
+            if new_block["count"]["max"] > len(new_block["items"]):
+                count = new_block["count"]["max"]
+                failed(f"Plando count {count} greater than items specified", block.force)
+                new_block["count"]["max"] = len(new_block["items"])
+                if new_block["count"]["min"] > len(new_block["items"]):
+                    new_block["count"]["min"] = len(new_block["items"])
+            if new_block["count"]["max"] > len(new_block["locations"]) > 0:
+                count = new_block["count"]["max"]
+                failed(f"Plando count {count} greater than locations specified", block.force)
+                new_block["count"]["max"] = len(new_block["locations"])
+                if new_block["count"]["min"] > len(new_block["locations"]):
+                    new_block["count"]["min"] = len(new_block["locations"])
+            new_block["count"]["target"] = multiworld.random.randint(new_block["count"]["min"],
```

Comment:
Yeah, I'm really starting to think that "PlandoBlock" should be a dataclass or namedtuple or something
Especially if we are expecting ER to be able to "look at it" and derive useful info from it